File: 151-160/153. Find Minimum in Rotated Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function (nums) {
    let min = nums[0];
    let len = nums.length;
    let left = 0, right = len - 1;
    while(left < right){
        let mid = left + Math.floor((right-left+1)/2);
        if(nums[mid]>min){
            left = mid;
        }
        else{
            right = mid - 1;
            min = nums[mid];
        }
    }
    return min;
};

File: 151-160/153. Find Minimum in Rotated Sorted Array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        first = nums[0]
        minVal = first

        l, r = 0, len(nums) - 1

        while l <= r:
            mid = l + (r - l) // 2
            if nums[mid] >= first:
                l = mid + 1
            elif nums[mid] < first:
                r = mid - 1
                minVal = min(minVal, nums[mid])

        return minVal
